tests/network/probes.go

```diff
@@ -103,29 +102,54 @@ var _ = SIGDescribe("[ref_id:1182]Probes", func() {
-
-			if disableEnableCycle {
-				By("Disabling the guest-agent")
-				Expect(console.LoginToFedora(vmi)).To(Succeed())
-				Expect(stopGuestAgent(vmi)).ToNot(HaveOccurred())
-
-				// Marking the status to not ready can take a little more time
-				tests.WaitForVMIConditionRemovedOrFalse(virtClient, vmi, v1.VirtualMachineInstanceReady, 300)
-
-				By("Enabling the guest-agent again")
-				Expect(startGuestAgent(vmi)).ToNot(HaveOccurred())
-				tests.WaitForVMICondition(virtClient, vmi, v1.VirtualMachineInstanceReady, 120)
-			}
 		},
-			Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server on ipv4", tcpProbe, corev1.IPv4Protocol, false),
-			Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server on ipv6", tcpProbe, corev1.IPv6Protocol, false),
-			Entry("[test_id:1200][posneg:positive]with working HTTP probe and http server on ipv4", httpProbe, corev1.IPv4Protocol, false),
-			Entry("[test_id:1200][posneg:positive]with working HTTP probe and http server on ipv6", httpProbe, corev1.IPv6Protocol, false),
-			Entry("[test_id:TODO]with working Exec probe", createExecProbe(period, initialSeconds, timeoutSeconds, "uname", "-a"), blankIPFamily, false),
-			Entry("[test_id:6739]with GuestAgentPing", guestAgentPingProbe, blankIPFamily, false),
-			Entry("[test_id:6741]status change with guest-agent availability", guestAgentPingProbe, blankIPFamily, true),
+			Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server on ipv4", tcpProbe, corev1.IPv4Protocol),
+			Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server on ipv6", tcpProbe, corev1.IPv6Protocol),
+			Entry("[test_id:1200][posneg:positive]with working HTTP probe and http server on ipv4", httpProbe, corev1.IPv4Protocol),
+			Entry("[test_id:1200][posneg:positive]with working HTTP probe and http server on ipv6", httpProbe, corev1.IPv6Protocol),
+			Entry("[test_id:TODO]with working Exec probe", createExecProbe(period, initialSeconds, timeoutSeconds, "uname", "-a"), blankIPFamily),
 		)
 
+		Context("guest agent ping", func() {
+			BeforeEach(func() {
+				vmi = tests.NewRandomFedoraVMIWithGuestAgent()
+				vmi.Spec.ReadinessProbe = createGuestAgentPingProbe(period, initialSeconds)
+				vmi = tests.VMILauncherIgnoreWarnings(virtClient)(vmi)
+				By("Waiting for agent to connect")
+				tests.WaitAgentConnected(virtClient, vmi)
+				tests.WaitForVMICondition(virtClient, vmi, v1.VirtualMachineInstanceReady, 120)
+			})
+
+			When("the guest agent is disabled", func() {
+				It("the VMI stops being ready", func() {
+					By("Disabling the guest-agent")
+					Expect(console.LoginToFedora(vmi)).To(Succeed())
+					Expect(stopGuestAgent(vmi)).ToNot(HaveOccurred())
+
+					// Marking the status to not ready can take a little more time
+					tests.WaitForVMIConditionRemovedOrFalse(virtClient, vmi, v1.VirtualMachineInstanceReady, 300)
+				})
+			})
+
+			When("the guest agent is enabled, after being disabled", func() {
```

Comment:
If I understand correctly, with this change we are running yet another Fedora VM. I feel like we should be more careful with running more VMs: starting them takes time and adds fragility to our already-brittle test framework.

I think that in this context it would be better to start a VM, ensure that it is non-ready when the agent is off, and that it becomes ready when the agent is on - all in the same VM.

Reply:
Hard to argue against reason. I'll factor this in. I'll make sure to also copy the `testid`s from the original table, to preserve traceability on our downstream tests.

Reply:
Done. I've removed the extra test, and moved the appropriate code to the test setup stage.

Reply:
I'm not sure that the `When`s make it easier for me. To me it would be easier to read a linear
```
By("stopping agent")
By("starting agent")
It("should be Ready again")
```

Reply:
afaiu, the ginkgo `When` clauses are meant to make this explicit. `By` clauses are just a crutch. 

This will read in the execution log:
`When the guest agent is enabled, after being disabled It [test_id:6741] the VMI enters `Ready` state once again` 

... which could be improved.